Makefile

```diff
@@ -41,22 +41,26 @@ licenses: ## Verifies dependency licenses and requires GITHUB_TOKEN to be set
 	golicense hack/license-config.hcl karpenter
 
 apply: ## Deploy the controller into your ~/.kube/config cluster
-	$(WITH_GOFLAGS) ko apply -B -f config
+	helm template karpenter ./config/ | $(WITH_GOFLAGS) ko apply -B -f -
 
 delete: ## Delete the controller from your ~/.kube/config cluster
-	ko delete -f config
+	helm template karpenter ./config/ |  ko delete -f -
 
 codegen: ## Generate code. Must be run if changes are made to ./pkg/apis/...
 	./hack/codegen.sh
 
 publish: ## Generate release manifests and publish a versioned container image.
 	@aws ecr-public get-login-password --region us-east-1 | docker login --username AWS --password-stdin $(RELEASE_REPO)
-	$(WITH_RELEASE_REPO) $(WITH_GOFLAGS) ko resolve -B -t $(RELEASE_VERSION) --platform all -f config > $(RELEASE_MANIFEST)
+	@mkdir -p $(RELEASE_CHART)/karpenter/templates
+	@cp config/templates/* $(RELEASE_CHART)/karpenter/templates
+	$(WITH_RELEASE_REPO) $(WITH_GOFLAGS) ko resolve -B -t $(RELEASE_VERSION) --platform all -f config/values.yaml > $(RELEASE_CHART)/karpenter/values.yaml
+	yq e -i '.version = "$(RELEASE_VERSION)"' ./$(RELEASE_CHART)/karpenter/Chart.yaml
 
-helm: ## Generate Helm Chart
-	cp $(RELEASE_MANIFEST) charts/karpenter/templates
-	yq e -i '.version = "$(RELEASE_VERSION)"' ./charts/karpenter/Chart.yaml
-	cd charts; helm package karpenter; helm repo index .
+verifiy-helm:
+	helm lint $(RELEASE_CHART)/karpenter
```

Comment:
This is awesome. Do you mind adding this step to the `helm` target instead of making a new step?

Reply:
Agreed